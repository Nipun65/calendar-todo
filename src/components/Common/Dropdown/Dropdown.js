import { useState } from 'react';
import styles from './Dropdown.module.css';
import Button from '../../UI/Button/Button';

function Dropdown({ option, setValue }) {
  const [showMenu, setShowMenu] = useState(false);
  const [selectedOption, setSelectedOption] = useState('Month');

  // handling selected option
  const handleOptionClick = (value) => {
    setShowMenu(!showMenu);
    setSelectedOption(value.target.textContent);
    setValue(value.target.textContent);
  };

  return (
    <div data-testid="dropdown">
      <Button
        onClick={() => setShowMenu(!showMenu)}
        classes="button is-primary"
        textContent={
          <div className={`${styles.textContent}`}>
            <span>{selectedOption}</span>
            <span className="icon is-small">
              <i className="fas fa-angle-down" aria-hidden="true" />
            </span>
          </div>
        }
      />

      {showMenu ? (
        <ul className={styles['dropdown-menu']}>
          {option.map((value) => (
            <li
              role="presentation"
              onClick={handleOptionClick}
              key={Math.random()}
              className={`has-text-centered ${styles.dropdownli} is-clickable`}
            >
              {value}
            </li>
          ))}
        </ul>
      ) : null}
    </div>
  );
}
export default Dropdown;
