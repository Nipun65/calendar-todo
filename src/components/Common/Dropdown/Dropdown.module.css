ul {
  position: absolute;
  border: 1px solid gray;
}

.dropdown-menu {
  width: 100px;
}

.dropdown-menu li {
  padding: 12px 16px;
}

.dropdownli:hover {
  background-color: #00d1b2;
}

.textContent :hover {
  background-color: none !important;
}

@media only screen and (max-width: 1200px) {
  .titlebutton {
    width: 122px;
  }
  .dropdown-menu {
    width: 122px;
  }
}

@media only screen and (max-width: 900px) {
  .titlebutton {
    width: 92px;
  }
  .dropdown-menu {
    width: 92px;
  }
}

@media only screen and (max-width: 600px) {
  .titlebutton {
    width: 82px;
  }
  .dropdown-menu {
    width: 82px;
  }
}
