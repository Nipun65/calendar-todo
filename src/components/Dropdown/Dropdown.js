import { useState } from "react";
import styles from "./Dropdown.module.css";
import Button from "../Button/Button";
const Dropdown = ({ option, setValue }) => {
  const [showMenu, setShowMenu] = useState(false);
  const [selectedOption, setSelectedOption] = useState("Month");
  const handleOptionClick = (value) => {
    setShowMenu(!showMenu);
    setSelectedOption(value.target.textContent);
    setValue(value.target.textContent);
  };
  return (
    <div className={styles.dropdown}>
      <Button
        onClick={() => setShowMenu(!showMenu)}
        classes={styles.titlebutton}
        textContent={selectedOption}
      />

      {showMenu ? (
        <ul className={styles["dropdown-menu"]}>
          {option.map((value) => (
            <li onClick={handleOptionClick} key={Math.random()}>
              {value}
            </li>
          ))}
        </ul>
      ) : null}
    </div>
  );
};
export default Dropdown;
