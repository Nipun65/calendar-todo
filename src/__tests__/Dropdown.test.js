import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Dropdown from '../components/Common/Dropdown/Dropdown';

afterEach(() => {
  cleanup();
});

it('should match the snapshot', () => {
  const tree = renderer.create(<Dropdown />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render dropdown Component', () => {
  const DROPDOWNLIST = ['Month', 'Year'];
  render(<Dropdown option={DROPDOWNLIST} />);
  const mainDiv = screen.getByTestId('dropdown');
  expect(mainDiv).toBeInTheDocument();
  expect(mainDiv).toHaveTextContent(DROPDOWNLIST[0]);
});
