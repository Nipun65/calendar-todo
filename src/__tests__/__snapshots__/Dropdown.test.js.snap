// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <button
    className="button is-primary is-clickable"
    onClick={[Function]}
    type="button"
  >
    <div
      className="textContent"
    >
      <span>
        Month
      </span>
      <span
        className="icon is-small"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </div>
  </button>
</div>
`;
